webapp/loggedin/appView.js

```diff
@@ -1,28 +1,46 @@
 import './appView.scss'
 
 import React from 'react'
+import { connect } from 'react-redux'
+
+import AppContext from '../app/appContext'
+
+import * as AppState from '../app/appState'
 
 import AppSideBar from './appSideBar/appSideBar'
 import AppJobMonitor from './appJob/appJobMonitor'
 import AppErrors from './appErrors/appErrors'
 import ModuleSwitch from './modules/moduleSwitch'
 
-const AppView = (props) => {
-  const pathName = props.history.location.pathname
+class AppView extends React.Component {
+  render () {
+    const { location, i18n } = this.props
+
+    const pathName = this.props.history.location.pathname
 
-  return (
-    <div className="app__container">
+    return (
+      <div className="app__container">
+        <AppContext.Provider value={{ i18n }}>
 
-      <AppSideBar pathname={pathName}/>
+          <AppSideBar pathname={pathName}/>
 
-      <ModuleSwitch location={props.location}/>
+          <ModuleSwitch location={location}/>
 
-      <AppJobMonitor/>
+          <AppJobMonitor/>
 
-      <AppErrors/>
+          <AppErrors/>
+
+        </AppContext.Provider>
+
+      </div>
+    )
+  }
+}
 
-    </div>
-  )
+const mapStateToProps = (state) => {
+  return { i18n: AppState.getState(state).i18n }
```

Comment:
Let's create a method in AppState.getI18n?

Reply:
ok